app/src/main/java/com/dessalines/thumbkey/ui/components/keyboard/KeyboardKey.kt

```diff
@@ -173,30 +178,72 @@ fun KeyboardKey(
                         val (x, y) = dragAmount
                         offsetX += x
                         offsetY += y
+                        if (key.slideType == SlideType.MOVE_CURSOR && spacebarSlide) {
+                            if (abs(offsetX) > slideSensitivity) {
+                                var direction = 0
+                                var shouldMove = false
+                                if (offsetX < 0.00) {
+                                    // move left
+                                    if (ime.currentInputConnection.getTextBeforeCursor(1, 0)?.length != 0) {
+                                        shouldMove = true
+                                    }
+                                    direction = KeyEvent.KEYCODE_DPAD_LEFT
+                                } else {
+                                    // move right
+                                    if (ime.currentInputConnection.getTextAfterCursor(1, 0)?.length != 0) {
+                                        shouldMove = true
+                                    }
+                                    direction = KeyEvent.KEYCODE_DPAD_RIGHT
+                                }
+                                if (shouldMove) {
+                                    val action = KeyAction.SendEvent(
+                                        KeyEvent(
+                                            KeyEvent.ACTION_DOWN,
+                                            direction,
+                                        ),
+                                    )
+                                    performKeyAction(
+                                        action = action,
+                                        ime = ime,
+                                        autoCapitalize = autoCapitalize,
+                                        onToggleShiftMode = onToggleShiftMode,
+                                        onToggleNumericMode = onToggleNumericMode,
+                                        onToggleCapsLock = onToggleCapsLock,
+                                        onAutoCapitalize = onAutoCapitalize,
+                                        onSwitchLanguage = onSwitchLanguage,
+                                        onSwitchPosition = onSwitchPosition,
+                                    )
+                                }
+                                offsetX = 0f
+                                offsetY = 0f
+                            }
+                        }
```

Comment:
This worked great! The only issue I can see, is that after sliding, the spacebar key appears pressed, and not released. If you can't figure that one out, its no big deal, we can fix that later.